Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/tangential_relaxation.h

```diff
@@ -0,0 +1,321 @@
+*                    as key type and `%Point_3` as value type}
+*     \cgalParamDefault{`boost::get(CGAL::vertex_point, tm)`}
+*     \cgalParamExtra{If this parameter is omitted, an internal property map for `CGAL::vertex_point_t`
+*                     must be available in `PolygonMesh`.}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{geom_traits}
+*     \cgalParamDescription{an instance of a geometric traits class}
+*     \cgalParamType{a class model of `Kernel`}
+*     \cgalParamDefault{a \cgal Kernel deduced from the `Point_3` type, using `CGAL::Kernel_traits`}
+*     \cgalParamExtra{The geometric traits class must be compatible with the vertex `Point_3` type.}
+*     \cgalParamExtra{Exact constructions kernels are not supported by this function.}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{number_of_iterations}
+*     \cgalParamDescription{the number of iterations for the sequence of atomic operations performed (listed in the above description)}
+*     \cgalParamType{unsigned int}
+*     \cgalParamDefault{`1`}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{edge_is_constrained_map}
+*     \cgalParamDescription{a property map containing the constrained-or-not status of each edge of `tm`}
+*     \cgalParamType{a class model of `ReadWritePropertyMap` with `boost::graph_traits<PolygonMesh>::%edge_descriptor`
+*                    as key type and `bool` as value type. It must be default constructible.}
+*     \cgalParamDefault{a default property map where no edge is constrained}
+*     \cgalParamExtra{A constrained edge can be split or collapsed, but not flipped, nor its endpoints moved by smoothing.}
+*     \cgalParamExtra{Sub-edges generated by splitting are set to be constrained.}
+*     \cgalParamExtra{Patch boundary edges (i.e. incident to only one face in the range) are always considered as constrained edges.}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{vertex_is_constrained_map}
+*     \cgalParamDescription{a property map containing the constrained-or-not status of each vertex of `tm`.}
+*     \cgalParamType{a class model of `ReadWritePropertyMap` with `boost::graph_traits<PolygonMesh>::%vertex_descriptor`
+*                    as key type and `bool` as value type. It must be default constructible.}
+*     \cgalParamDefault{a default property map where no vertex is constrained}
+*     \cgalParamExtra{A constrained vertex cannot be modified during remeshing.}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{relax_constraints}
+*     \cgalParamDescription{If `true`, the end vertices of the edges set as constrained
+*                           in `edge_is_constrained_map` and boundary edges move along the}
+*                           constrained polylines they belong to.}
+*     \cgalParamType{Boolean}
+*     \cgalParamDefault{`false`}
+*   \cgalParamNEnd
+*
+* \cgalNamedParamsEnd
+*
+* \todo check if it should really be a triangle mesh or if a polygon mesh is fine
```

Comment:
Currently `iso_remesh()` uses `PolygonMesh` with "triangulated surface patches"; `angle_and_area_smoothing()` requires explicitly triangle meshes. Maybe `isotropic_remeshing()` should be changed to `TriangleMesh` for consistency.

Reply:
Right. In `isotropic_remeshing()`, only the to-be-remeshed faces must be triangle faces, and imho it's a feature we don't want to give up, for example for hybrid meshes.

Actually tangential relaxation does not have such requirements and should work on any polygon mesh, and (if I'm right) I think it would be a pity to restrict it to triangle meshes if it's not a requirement.